decoder.go

```diff
@@ -130,9 +130,12 @@ func (d *Decoder) setDefaults(t reflect.Type, v reflect.Value) MultiError {
 				defaultSlice := reflect.MakeSlice(f.typ, 0, cap(vals))
 				for _, val := range vals {
 					//this check is to handle if the wrong value is provided
-					if convertedVal := builtinConverters[f.typ.Elem().Kind()](val); convertedVal.IsValid() {
-						defaultSlice = reflect.Append(defaultSlice, convertedVal)
+					convertedVal := builtinConverters[f.typ.Elem().Kind()](val)
+					if !convertedVal.IsValid() {
+						errs.merge(MultiError{"default-" + f.name: errors.New("some elements in the slice are of a different type.")})
```

Comment:
I suggest some changes to the error message:
* first of all an error message should not end with dot: https://staticcheck.dev/docs/checks/#ST1005
* it would be better if the error message mentions that the error happened during setting the default. For example: `failed setting default value for FIELD_NAME`
* It would make also the error more helpful if the error message mentions which value in the tag option is not compatible with the slice type

I am thinking of something like: `failed setting default: VALUE is not compatible with field FIELD_NANE type`